### PR TITLE
Clickable map points

### PR DESCRIPTION
**Pull request by rbelateche**
# HEX-XX Clickable map points

Task number **# [(HEX-)](https://hexanax.atlassian.net/browse/HEX-)**

## Description

I added the functionnality of clickable map intersections and requests when needed in case of adding or deleting a request.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Testing the changes**:

## Checklist:

- [ X] My code follows the style guidelines of this project
- [W ] I have performed a self-review of my own code
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: src/main/java/fr/insalyon/pldagile/controller/AddRequestState7.java [hunk 1]
```diff
@@ -10,6 +10,8 @@
     @Override
     public void addRequest(Controller controller, CityMap citymap, Tour tour, Tour modifyTour, Long intersectionID, Window window) {
         TourBuilderV2 tourBuilder = new TourBuilderV2();
+        System.out.println("pickupToAdd:" + controller.pickupToAdd);
+        System.out.println("deliveryToAdd:" + controller.deliveryToAdd);
```
Comment (bdeleglise) [inline src/main/java/fr/insalyon/pldagile/controller/AddRequestState7.java:14]: Delete Sys.out pls

**Review by bdeleglise — APPROVED** [review]

**Comment by M4TTRX** [comment]
Je te recommande de créer un task sur jira et d'associer cette pr à un task afin de pouvoir correctement suivre les charges du projet

File: src/main/java/fr/insalyon/pldagile/view/Window.java [hunk 2]
```diff
@@ -299,11 +306,30 @@ public void orderListRequests(ArrayList<Pair<Long, String>> steps, Map<Long, Req
             {
                 item = new RequestItem("Pickup at " + requests.get(step.getKey()).getPickup().getIntersection().getId(), "Duration: " + requests.get(step.getKey()).getPickup().getDuration(), step.getKey(), "Pickup",index);
                 items.add(item);
+
+                MapPoint mapPoint = new MapPoint(requests.get(step.getKey()).getPickup().getIntersection().getCoordinates().getLatitude(), requests.get(step.getKey()).getPickup().getIntersection().getCoordinates().getLongitude());
```
Comment (M4TTRX) [inline src/main/java/fr/insalyon/pldagile/view/Window.java:310]: pls spilt this line up its super long and hard to read

**Review by M4TTRX — CHANGES_REQUESTED** [review]
This doesn't work, you should probably detect the nearest intersection to the click within a reasonable distance, because it is very difficult to click exactly where the intersection is.

Or ideally if the user clicks on the intersection point

**Review by M4TTRX — APPROVED** [review]

File: src/main/java/fr/insalyon/pldagile/view/Window.java (hunk 2 above)
Comment (rbelateche) [reply src/main/java/fr/insalyon/pldagile/view/Window.java:310]: Done

File: src/main/java/fr/insalyon/pldagile/controller/AddRequestState7.java (hunk 1 above)
Comment (rbelateche) [reply src/main/java/fr/insalyon/pldagile/controller/AddRequestState7.java:14]: Done yall

Comment (rbelateche) [reply src/main/java/fr/insalyon/pldagile/controller/AddRequestState7.java]: Done yall

**Pull request merged**